packages/vscode-ui5-language-assistant/src/registerManifestSchema.ts

```diff
@@ -0,0 +1,104 @@
+import { workspace } from "vscode";
+import {
+  getManifestSchemaConfigEntry,
+  IJSONValidationExtensionPoint,
+  MANIFEST_COMMENT,
+  MANIFEST_FILE_MATCH,
+} from "./manifestSchemaConfig";
+
+/**
+ *  The configuration will be written in workspace.settings
+ */
+export function registerManifestSchema(): void {
+  const jsonSchemaConfig:
+    | IJSONValidationExtensionPoint
+    | undefined = workspace.getConfiguration().get("json.schemas");
+  if (isArray(jsonSchemaConfig)) {
+    const previousConfigEntryIndex = getIndexOfPreviousConfigEntry(
+      jsonSchemaConfig
+    );
+    if (previousConfigEntryIndex !== -1) {
+      updatePreviousConfigEntry(jsonSchemaConfig, previousConfigEntryIndex);
+    } else if (isMissingConfig(jsonSchemaConfig)) {
+      createConfig(jsonSchemaConfig);
+    } else {
+      // No change is needed
+    }
+  }
+}
+
+function isMissingConfig(
+  jsonSchemaConfig: IJSONValidationExtensionPoint[]
+): boolean {
+  const missingConfigEntry =
+    find(
+      jsonSchemaConfig,
+      (jsonSchemaConfigEntry: IJSONValidationExtensionPoint) => {
+        if (isArray(jsonSchemaConfigEntry.fileMatch)) {
+          return some(
+            jsonSchemaConfigEntry.fileMatch,
+            (fileMatchEntry) => fileMatchEntry === MANIFEST_FILE_MATCH
+          );
+        }
+
+        return jsonSchemaConfigEntry.fileMatch === MANIFEST_FILE_MATCH;
+      }
+    ) === undefined;
+
+  return missingConfigEntry;
+}
+
+function createConfig(jsonSchemaConfig: IJSONValidationExtensionPoint[]): void {
+  const configEntry = getManifestSchemaConfigEntry();
+  const updatedSchemaConfig = clone(jsonSchemaConfig);
+  updatedSchemaConfig.push(configEntry);
+
+  configureSchemaInWorkspaceSettings(updatedSchemaConfig);
+}
+
+function getIndexOfPreviousConfigEntry(
+  jsonSchemaConfig: IJSONValidationExtensionPoint[]
+): number {
+  const previousConfigEntryIndex = findIndex(
+    jsonSchemaConfig,
+    (jsonSchemaConfigEntry: IJSONValidationExtensionPoint) => {
+      if (
+        isArray(jsonSchemaConfigEntry.fileMatch) &&
+        jsonSchemaConfigEntry.fileMatch.length === 1
+      ) {
+        return some(
+          jsonSchemaConfigEntry.fileMatch,
+          (fileMatchEntry) =>
+            fileMatchEntry === MANIFEST_FILE_MATCH &&
+            jsonSchemaConfigEntry.comment === MANIFEST_COMMENT
+        );
+      }
+
+      return false;
+    }
+  );
+
+  return previousConfigEntryIndex;
+}
+
+function updatePreviousConfigEntry(
+  jsonSchemaConfig: IJSONValidationExtensionPoint[],
+  previusConfigEntryIndex: number
+): void {
+  const updatedSchemaConfig = clone(jsonSchemaConfig);
+  updatedSchemaConfig[previusConfigEntryIndex] = getManifestSchemaConfigEntry();
+
+  configureSchemaInWorkspaceSettings(updatedSchemaConfig);
+}
+
+function configureSchemaInWorkspaceSettings(
+  updatedSchemaConfig: IJSONValidationExtensionPoint[]
+): void {
+  // We pass 'undefined' for multi-root
```

Comment:
lets pass `true` to write to the global settings instead of the user's workspace.
> - If `true` updates [Global settings](#ConfigurationTarget.Global).

This will simplify state management as the configurations would be written in some root level `.theia` folder outside the user's workspace, so there would be no source control conflicts.